backend/src/firebase/user/user.service.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable prettier/prettier */
+import { Injectable } from '@nestjs/common';
+import { AdminService } from '../admin/admin.service';
+import { User } from 'src/interfaces/user.interface';
+import * as bcrypt from 'bcryptjs'
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class UserService {
+  constructor(private readonly adminService: AdminService, private readonly jwtService: JwtService){}
+
+  async registerUser(data: User){
+    const firestore = this.adminService.getFirestore();
+
+    try{
```

Comment:
Postaraj się zrobić coś takiego:
1) Utworzyć plik w jakimś folderze, który nazwiesz np. "helpers" z constem, który będzie zawierał wszystkie wiadomości błędów. A w serwisach potem używaj tych stałych. Dzięki temu będziesz miał to uporządkowane i jak będziesz chciał zmienić jakąś wiadomość to będziesz łatwo wiedział gdzie szukasz. Możliwe też że będziesz niektóre używał wiele razy. Ma to wyglądać mniej więcej tak:

export const ERROR_MESSAGES = {
loginErrors: {
usernameError: "This username is already in use",
repeatPasswordError: "Password and Repeat password is not match",
...
}
}